app/src/main/java/com/lftechnology/unito/activity/MainActivity.java

```diff
@@ -0,0 +1,312 @@
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mToolbarContainer = (LinearLayout) findViewById(R.id.toolbarContainer);
+        toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        setSpinner();
+        setNavigationDrawer();
+    }
+
+    private void setRecyclerView() {
+        mRecyclerView = (RecyclerView) findViewById(R.id.drawer_recycler_view);
+        mRecyclerView.setHasFixedSize(true);
+        mLayoutManager = new LinearLayoutManager(this);
+        mRecyclerView.setLayoutManager(mLayoutManager);
+        mDrawerRecyclerViewDataset = getDrawerRecyclerViewDataset();
+        mAdapter = new DrawerRecyclerViewAdapter(mDrawerRecyclerViewDataset, this, mSelectedConversion);
+        mRecyclerView.setAdapter(mAdapter);
+
+        ItemTouchHelper.Callback callback = new SimpleItemTouchHelperCallback(mAdapter);
+        mItemTouchHelper = new ItemTouchHelper(callback);
+        mItemTouchHelper.attachToRecyclerView(mRecyclerView);
+
+        linearLayout = (LinearLayout) findViewById(R.id.inflated_content_main);
+    }
+
+    private void setNavigationDrawer() {
+        mDrawerLayout = (DrawerLayout) findViewById(R.id.drawer_layout);
+        mDrawerToggle = setupDrawerToggle();
+        mDrawerLayout.addDrawerListener(mDrawerToggle);
+    }
+
+    private String[] getDrawerRecyclerViewDataset() {
+        String[] drawerRecyclerViewDataset;
+        SharedPreferences sharedPref = getBaseContext().getSharedPreferences("Unito", Context.MODE_PRIVATE);
+        String preference = sharedPref.getString(mSelectedConversion, "N/A");
+        if (preference.equals("N/A")) {
+            switch (mSelectedConversion) {
+                case AppConstant.LENGTH:
+                    drawerRecyclerViewDataset = getResources().getStringArray(R.array.length_options);
+                    break;
+                case AppConstant.TEMPERATURE:
+                    drawerRecyclerViewDataset = getResources().getStringArray(R.array.temperature_options);
+                    break;
+                case AppConstant.TIME:
+                    drawerRecyclerViewDataset = getResources().getStringArray(R.array.time_options);
+                    break;
+                case AppConstant.VOLUME:
+                    drawerRecyclerViewDataset = getResources().getStringArray(R.array.volume_options);
+                    break;
+                case AppConstant.WEIGHT:
+                    drawerRecyclerViewDataset = getResources().getStringArray(R.array.weight_options);
+                    break;
+                default:
+                    drawerRecyclerViewDataset = new String[]{};
+                    break;
+            }
+        } else {
+            Gson gson = new Gson();
+            String[] arrayList = gson.fromJson(preference, String[].class);
+            drawerRecyclerViewDataset = arrayList;
```

Comment:
@grishmashrestha drawerRecyclerViewDataset = gson.fromJson(preference, String[].class); i think this does the job too. why extra array initialization.